src/core/build/geometryUtils.js

```diff
@@ -73,18 +73,18 @@ export function getFullGeometryRange( geo ) {
 
 }
 
-export function getRootIndexRanges( geo ) {
+export function getRootIndexRanges( geo, range ) {
 
 	if ( ! geo.groups || ! geo.groups.length ) {
 
-		return getFullGeometryRange( geo );
+		return getFullGeometryRange( geo, range );
 
 	}
 
 	const ranges = [];
 	const rangeBoundaries = new Set();
 
-	const drawRange = geo.drawRange;
+	const drawRange = range ? getRangesIntersection( geo.drawRange, range ) : geo.drawRange; // or we just want to use only 'range' also here?
```

Comment:
I think we should just use range directly here instead of computing the overlap with "drawRange".